packages/xo-web/src/common/xo/index.js

```diff
@@ -1037,6 +1037,21 @@ export const convertVmToTemplate = vm =>
     ),
   }).then(() => _call('vm.convert', { id: resolveId(vm) }), noop)
 
+export const changeVirtualizationMode = vm =>
+  confirm({
+    title: 'Change Virtualization mode',
+    body: (
+      <div>
+        <p>
+          You must know what you are doing, because it could break your setup
+          (if you didn't installed the bootloader in the MBR while switching
+          from PV to HVM, or even worse, in HVM to PV, if you don't have the
+          correct PV args)
+        </p>
+      </div>
+    ),
+  }).then(() => editVm(vm, { virtualizationMode: vm.virtualizationMode }))
```

Comment:
Are you sure this is correct, it seems that you are setting the mode to *pv` if the VM is *pv*?

Reply:
No, actually, we send the current current mode. And then we will change the HVM_boot_policy according to the mode.

Reply:
That does not make sense, `editVm` is used to set properties to new values.

Reply:
And no need to pass the current value, because the server already knows it.